model-optimizer/extensions/front/tf/CTCGreedyDecoder.py

```diff
@@ -61,22 +70,49 @@ def replace_sub_graph(self, graph: Graph, match: dict):
-                        'supported with the Inference Engine.'.format(node.soft_get('name')))
-        # the batch size is the dimension with index 1 for the layer CTCGreedyDecoder
-        new_value = np.ones([node.in_node(0).shape[1], sequence_length_node.value[0]])
-        new_value[:, 0] = 0
-        new_value = np.transpose(new_value)
-        sequence_length_node.value = new_value
-        sequence_length_node.shape = int64_array(sequence_length_node.value.shape)
-
-        node.old_infer(node)
+        shape = Shape(graph, {'name': decoder_name + '/shape'}).create_node()
+
+        ss = create_op_with_const_inputs(graph, StridedSlice, {1: int64_array([1]),
+                                                               2: int64_array([2]),
+                                                               3: int64_array([1])},
+                                                               {'name': decoder_name + '/strided_slice',
+                                                                'begin_mask': '1',
+                                                                'ellipsis_mask': '0',
+                                                                'end_mask': '1',
+                                                                'new_axis_mask': '0',
+                                                                'shrink_axis_mask': '1'})
+
+        decoder_node.in_port(0).get_source().connect(shape.in_port(0))
+        shape.out_port(0).connect(ss.in_port(0))
+
+        unsqueeze_1 = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0, 1]),
+                                                       {'name': decoder_name + '/unsqueeze_1'})
+        unsqueeze_1.in_port(0).connect(ss.out_port(0))
+
+        unsqueeze_2 = create_op_node_with_second_input(graph, Unsqueeze, int64_array([0]),
+                                                       {'name': decoder_name + '/unsqueeze_2'})
+
+        port = decoder_node.in_port(1).get_source()
+        port.disconnect()
+        port.connect(unsqueeze_2.in_port(0))
+
+        concat = Concat(graph, {'name': decoder_name + '/concat'}).create_node()
+        concat.add_input_port(0, skip_if_exist=True)
+        concat.add_input_port(1, skip_if_exist=True)
+
+        concat.in_port(0).connect(unsqueeze_2.out_port(0))
+        concat.in_port(1).connect(unsqueeze_1.out_port(0))
+
+        broadcast = create_op_with_const_inputs(graph, Broadcast, {0: int64_array([1])},
+                                                {'name': decoder_name + '/broadcast'})
+
+        squeeze = create_op_node_with_second_input(graph, Squeeze, int64_array([0]),
+                                                   {'name': decoder_name + '/squeeze'})
+        concat.out_port(0).connect(squeeze.in_port(0))
+        squeeze.out_port(0).connect(broadcast.in_port(1))
+
+        broadcast.out_port(0).connect(decoder_node.in_port(1))
+
+        return {}
```

Comment:
do we really need to use FrontReplacementSubgraph here? FrontReplacementPattern would be enough. 
Lets handle `nodes_to_remove` business inside the `replace_pattern`.